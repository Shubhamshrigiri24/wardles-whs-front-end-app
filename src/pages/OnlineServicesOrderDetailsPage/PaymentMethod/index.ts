export { default, PaymentMethod } from "./PaymentMethod";
