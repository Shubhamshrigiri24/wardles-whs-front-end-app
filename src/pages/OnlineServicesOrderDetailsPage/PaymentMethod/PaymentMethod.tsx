import React, { FC } from "react";
import { useStyles as useSharedStyles } from "pages/OnlineServicesOrderDetailsPage/styles";
import { Typography, Alert } from "@welldigital/components";

export type PaymentMethodProps = {
  lastFour: string;
  braintreeToken: string | null | undefined;
  braintreeTokenErrorMessage: string;
  onChangePaymentMethod: () => void;
};

export const PaymentMethod: FC<PaymentMethodProps> = ({
  lastFour,
  braintreeToken,
  braintreeTokenErrorMessage,
  onChangePaymentMethod,
}) => {
  const classes = { ...useSharedStyles() };

  return (
    <>
      <Typography className={classes.subTitle} spacingAfter={1}>
        Payment method
      </Typography>
      {braintreeTokenErrorMessage && (
        <Alert
          type={"error"}
          spacingAfter={1}
          message={braintreeTokenErrorMessage}
          data-testid={"alert/message"}
        />
      )}
      {braintreeToken && (
        <>
          {lastFour && (
            <Typography
              className={classes.paragraph}
              spacingAfter={1}
              data-testid={"paymentMethod/lastFour"}
            >
              XXXX XXXX XXXX {lastFour}
            </Typography>
          )}

          <Typography
            className={classes.link}
            spacingAfter={4}
            onClick={onChangePaymentMethod}
            data-testid={"paymentMethod/change"}
          >
            Change your payment method
          </Typography>
        </>
      )}
    </>
  );
};

export default PaymentMethod;
