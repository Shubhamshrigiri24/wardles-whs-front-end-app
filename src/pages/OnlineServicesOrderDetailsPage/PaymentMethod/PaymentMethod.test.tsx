import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import PaymentMethod, { PaymentMethodProps } from "./PaymentMethod";

beforeEach(() => {
  jest.clearAllMocks();
});

const LAST_FOUR = "1111";
const braintreeToken = "12345678910";
const mockOnchangePaymentMethod = jest.fn();

const makeProps = (props?: Partial<PaymentMethodProps>): PaymentMethodProps => {
  return {
    lastFour: LAST_FOUR,
    braintreeToken,
    braintreeTokenErrorMessage: "",
    onChangePaymentMethod: mockOnchangePaymentMethod,
    ...props,
  };
};
describe("<PaymentMethod />", () => {
  it("should match snapshot", () => {
    const props = makeProps();
    const { container } = render(<PaymentMethod {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("should render payment method", () => {
    const props = makeProps();
    const { getByTestId } = render(<PaymentMethod {...props} />);
    expect(getByTestId("paymentMethod/lastFour")).toHaveTextContent(
      `XXXX XXXX XXXX ${LAST_FOUR}`
    );
    expect(getByTestId("paymentMethod/change")).toHaveTextContent(
      "Change your payment method"
    );
  });

  it("should render error", () => {
    const errorMessage = "Braintree error";
    const props = makeProps({
      braintreeToken: null,
      braintreeTokenErrorMessage: errorMessage,
    });
    const { getByTestId } = render(<PaymentMethod {...props} />);
    expect(getByTestId("alert/message")).toHaveTextContent(errorMessage);
  });

  it("should call onChangePaymentMethod when change is clicked", () => {
    const props = makeProps();
    const { getByTestId } = render(<PaymentMethod {...props} />);
    const changeButton = getByTestId("paymentMethod/change");
    fireEvent.click(changeButton);
    expect(mockOnchangePaymentMethod).toHaveBeenCalledTimes(1);
  });
});
