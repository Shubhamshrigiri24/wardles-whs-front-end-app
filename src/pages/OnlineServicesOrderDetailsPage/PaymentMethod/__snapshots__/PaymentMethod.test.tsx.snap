// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentMethod /> should match snapshot 1`] = `
<div>
  <p
    class="MuiTypography-root makeStyles-subTitle-2 MuiTypography-body1"
  >
    Payment method
  </p>
  <div
    class="makeStyles-spacing1-14"
    data-testid="spacing/root"
  />
  
  <p
    class="MuiTypography-root makeStyles-paragraph-3 MuiTypography-body1"
    data-testid="paymentMethod/lastFour"
  >
    XXXX XXXX XXXX 
    1111
  </p>
  <div
    class="makeStyles-spacing1-14"
    data-testid="spacing/root"
  />
  <p
    class="MuiTypography-root makeStyles-link-7 MuiTypography-body1"
    data-testid="paymentMethod/change"
  >
    Change your payment method
  </p>
  <div
    class="makeStyles-spacing4-20"
    data-testid="spacing/root"
  />
</div>
`;
